src/psyclone/gocean1p0.py

```diff
@@ -963,11 +963,17 @@ def reference_accesses(self, var_accesses):
                     field_for_grid_property = \
                         self._arguments.find_grid_access()
                 var_name = arg.dereference(field_for_grid_property.name)
+                print("XXX", arg.type, 
+                    field_for_grid_property.name,
+                    arg.dereference(field_for_grid_property.name))
+
             else:
                 var_name = arg.name
 
             if arg.is_scalar():
-                var_accesses.add_access(var_name, arg.access, self)
+                # The argument is only a variable if it is not a constant:
```

Comment:
Yes, nicer, but I think incorrect:: if your suggested code handles a scalar literal, it would try to declare this literal as an array in the else branch.

Reply:
Pah, that's a small point. Beauty before correctness :-)